pkg/lock/dir.go

```diff
@@ -147,6 +149,11 @@ func NewLock(dir string) (*DirLock, error) {
 	// we can't use os.OpenFile as Go sets O_CLOEXEC
 	lfd, err := syscall.Open(l.dir, syscall.O_RDONLY|syscall.O_DIRECTORY, 0)
 	if err != nil {
+		if err == syscall.ENOENT {
```

Comment:
Is that really appropriate for comparison against an error out of syscall.Open though?  os.Open sure...